src/converters/construction_converter.py

```
@@ -0,0 +1,78 @@
+            return
+            
+        try:
+            self.logger.info(f"Adding Construction '{data.Name}' to IDF.")
+           
+            for layer_name in data.Layers:
+                if not (self.idf.getobject("MATERIAL", layer_name) or 
+                        self.idf.getobject("MATERIAL:NOMASS", layer_name)):
+                    raise ValueError(
+                        f"Material '{layer_name}' referenced in Construction '{data.Name}' "
+                        f"does not exist in IDF. Please add the material first."
+       )
```

Ruff (0.14.2)

Abstract `raise` to an inner function

(TRY301)